### PR TITLE
Update build process to sign mac releases

### LINKED ISSUE #243: Auto Update?
### Your issue

Hey guys,

Great work on Ferdium! Just switched over and love it! I see there's pretty regular Nightly updates - but they don't seem to be passed through via the Auto Updater on the actual app?

Is that intended / a result of the missing developer license? Please let me know!

All the best,
Oskar

### PR DESCRIPTION
**Pull request by Alphrag**
#### Pre-flight Checklist

1. Please remember that if you are logging a bug for some service that has _stopped working_ or is _working incorrectly_, please log the bug [here](https://github.com/ferdium/ferdium-recipes/issues)
2. If you are requesting support for a **new service** in Ferdium, please log it [here](https://github.com/ferdium/ferdium-recipes/pulls)
3. Please remember to read the [self-help documentation](https://github.com/ferdium/ferdium-app#troubleshooting-recipes-self-help) - in case it helps you unblock yourself for issues related to older versions of recipes that were installed on your machine. (These will get automatically upgraded when you upgrade to the newer versions of Ferdium, but to get new recipes between Ferdium releases, this documentation is quite useful.)
4. Please ensure you've completed all of the following.

- [x] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/develop/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://github.com/ferdium/ferdium-app/issues) for a feature request that matches the one I want to file, without success.

#### Description of Change

Add code signing and notarizing capabilities to the build for mac. 
This closes #243 as well as allowing natural resolution of #51 and #252.

#### Motivation and Context

Having the builds signed with a approved Apple certificate allows user to install the software without warnings, as well as permitting the auto-update to occur.

#### Checklist

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`npm run prepare-code`)
- [x] I tested/previewed similar changes on a personal fork

#### Release Notes

notes: Signed MacOS build.

### CODE REVIEW

File: build-helpers/notarize.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ exports.default = async function notarizing(context) {
   await notarize({
     appBundleId: 'org.ferdium.ferdium-app',
     appPath: `${appOutDir}/${appName}.app`,
-    ascProvider: 'B6J9X9DWFL',
+    ascProvider: '55E9FPJ93P',
```
Comment (SpecialAro) [inline build-helpers/notarize.js:15]: I'm imagining that this doesn't required to be a secret right?

**Review by SpecialAro — APPROVED** [review]
Looks perfect to me!!! 😁 Thank you so much @Alphrag for working hard on this!

File: build-helpers/notarize.js (hunk 1 above)
Comment (vraravam) [reply build-helpers/notarize.js:15]: no - this is good.

**Review by vraravam — APPROVED** [review]

File: build-helpers/notarize.js (hunk 1 above)
Comment (Alphrag) [reply build-helpers/notarize.js:15]: It appears in the certificate anyway if you look at it closely.

**Comment by vraravam** [comment]
@all-contributors please add @Alphrag for infra

**Pull request merged**